koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -945,3 +954,271 @@ def handler(self, src, target, opts=None):
+            },
+            {
+                "type": ["object"],
+                "properties": {
+                    "scratch": {
+                        "type": "boolean",
+                        "description": "Perform a scratch build?"
+                    },
+                    "koji_build_id": {
+                        "type": ["integer"],
+                        "description": "Koji build id for sources"
+                    },
+                    "koji_build_nvr": {
+                        "type": ["string"],
+                        "description": "Koji build nvr for sources"
+                    },
+                    "signing_intent": {
+                        "type": ["string"],
+                        "description": "Signing intent of the ODCS composes."
+                    }
+                },
+                "oneOf": [{
+                    "required": ["koji_build_nvr"]
+                }, {
+                    "required": ["koji_build_id"]
+                }],
+                "additionalProperties": "false"
+            }
+        ],
+        'minItems': 2
+    }
+
+    def __init__(self, id, method, params, session, options, workdir=None, demux=False):
+        BaseContainerTask.__init__(self, id, method, params, session, options,
+                                   workdir)
+        self._osbs = None
+        self.demux = demux
+
+        self._log_handler_added = False
+
+    def runBuilds(self, target_info, scratch=False, component=None, koji_build_id=None,
+                  koji_build_nvr=None, signing_intent=None):
+        self.logger.debug("Spawning job for sources")
+
+        kwargs = dict(
+            target_info=target_info,
+            scratch=scratch,
+            component=component,
+            koji_build_id=koji_build_id,
+            koji_build_nvr=koji_build_nvr,
+            signing_intent=signing_intent,
+        )
+
+        results = []
+        semi_results = self.createSourceContainer(**kwargs)
+        if semi_results is not None:
+            results = [semi_results]
+
+        self.logger.debug("Results: %r", results)
+        return results
+
+    def createSourceContainer(self, target_info=None, scratch=None, component=None,
+                              koji_build_id=None, koji_build_nvr=None, signing_intent=None):
+        this_task = self.session.getTaskInfo(self.id)
+        self.logger.debug("This task: %r", this_task)
+        owner_info = self.session.getUser(this_task['owner'])
+        self.logger.debug("Started by %s", owner_info['name'])
+
+        create_build_args = {
+            'user': owner_info['name'],
+            'component': component,
+            'sources_for_koji_build_id': koji_build_id,
+            'sources_for_koji_build_nvr': koji_build_nvr,
+            'target': target_info['name'],
+            'scratch': scratch,
+            'koji_task_id': self.id,
+        }
+
+        try:
+            source_create_build_args = create_build_args.copy()
```

Comment:
Any need to copy it? We only use it once.

Reply:
yeah for sourcecontainer we don't need copy since we use it just once